docs/references/function_calling.md

```diff
@@ -0,0 +1,155 @@
+# Tools Calling
+
+Currently, we added the support for tools calling in the following models:
+  - Llama 3.2 models
+  - Llama 3.1 models
+  - Qwen 2.5 models
+  - InternLM Models
+
+For adding support of more different architectures: \n
```

Comment:
Reorganize this. Mention the usage first and then mention how to support a new model.

```
# Supported models
# Usage
# How to support a new model
```